src/main/java/net/minecraftforge/datafix/ForgeDataFixerCustomFixersModStateTransition.java

```diff
@@ -0,0 +1,220 @@
+     */
+    public static IModStateTransition get() {
+        return new ForgeSerialModStateTransition<>(
+          ForgeDataFixerCustomFixersModStateTransition::generateFixersEvents,
+          ForgeDataFixerCustomFixersModStateTransition::preDispatchFixerEvents,
+          ForgeDataFixerCustomFixersModStateTransition::postDispatchFixerEvents,
+          ForgeDataFixerCustomFixersModStateTransition::rebuildRulesAfterSchemaEvents
+        );
+    }
+
+    /**
+     * Gives access to the list of schemas registered with the data fixers.
+     * If we are currently running an incompatible version of DFU or somebody messed with it,
+     * then an empty list is returned.
+     *
+     * @return The list of all known schemas to handle.
+     */
+    private static List<ForgeSchema> getAllSchemas()
+    {
+        //Check if we are running in a compatible environment.
+        if (DataFixers.getDataFixer() instanceof ForgeDataFixerDelegateHandler forgeDataFixerDelegateHandler) {
+            //Forge DFU wrapper is active, grab its configured schemas.
+            return forgeDataFixerDelegateHandler.getAllSchemas();
+        }
+
+        //Incompatible environment, return an empty list.
+        return Collections.emptyList();
+    }
+
+    /**
+     * Builder function for creating the events that are fired on start up.
+     * This returns one generator per schema, and keeps the schema in the order defined.
+     *
+     * The returned generator creates one event internally per schema and mod container, and then stores them for reuse.
+     *
+     * @return The stream of event generators used to fire one event per mod per schema.
+     */
+    private static Stream<IModStateTransition.EventGenerator<?>> generateFixersEvents()
+    {
+        //Grab the schemas
+        List<ForgeSchema> schemas = Lists.newArrayList(getAllSchemas());
+        //Sort them based on their version key.
+        schemas.sort(Comparator.comparingInt(Schema::getVersionKey));
+
+        //Create one generator per schema.
+        return schemas.stream().map(InternalEventGenerator::new);
+    }
+
+    /**
+     * Invoked by the event management system to indicate that we are about the fire an event for the given generator to all mods.
+     * This is invoked exactly once, and only once, per generator (and as such per event, and schema).
+     *
+     * Clears out the current status of the schema by resetting its type definition maps.
+     *
+     * @param executor The executor to execute this task on.
+     * @param eventGenerator The event generator that is about to be executed.
+     * @return A future that indicates success or failure.
+     */
+    private static CompletableFuture<List<Throwable>> preDispatchFixerEvents(
+      final Executor executor, final IModStateTransition.EventGenerator<RegisterFixesEvent> eventGenerator
+    )
+    {
+        if (!(eventGenerator instanceof InternalEventGenerator internalEventGenerator)) {
```

Comment:
Formatting: braces on newlines.
```suggestion
        if (!(eventGenerator instanceof InternalEventGenerator internalEventGenerator))
        {
```